plugins/@grouparoo/pipedrive/src/lib/export/exportProfile.ts

```
@@ -0,0 +1,139 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import { PipedriveClient } from "../client";
+import { connect } from "../connect";
+import {
+  PipedriveCacheData,
+  getKnownPersonFieldMap,
+} from "./destinationMappingOptions";
+import { getGroupFieldKey } from "./listMethods";
+
+export const exportProfile: ExportProfilePluginMethod = async ({
+  appId,
+  appOptions,
+  export: {
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+    toDelete,
+  },
+}) => {
+  if (Object.keys(newProfileProperties).length === 0) {
+    return { success: true };
+  }
+
+  const client = await connect(appOptions);
+  const cacheData: PipedriveCacheData = { appId, appOptions };
+
+  const newEmail = newProfileProperties["Email"];
+  const oldEmail = oldProfileProperties["Email"];
+
+  const newFoundId = await client.findPersonIdByEmail(newEmail);
+
+  let oldFoundId = null;
+  if (oldEmail && oldEmail !== newEmail) {
+    oldFoundId = await client.findPersonIdByEmail(oldEmail);
+  }
+
+  const foundId = oldFoundId || newFoundId;
+  if (toDelete) {
+    if (!foundId) {
+      throw new Error(
+        `Person with email ${oldEmail || newEmail} was not found to delete.`
```

In other ones, this fails silently, so probably `return` here